pkg/operator/ceph/pool/validate.go

```diff
@@ -163,3 +173,37 @@ func ValidatePoolSpec(context *clusterd.Context, clusterInfo *cephclient.Cluster
 
 	return nil
 }
+
+// validateDeviceClasses validates the primary and secondary device classes in the HybridStorageSpec
+func validateDeviceClasses(context *clusterd.Context, clusterInfo *cephclient.ClusterInfo,
+	p *cephv1.PoolSpec) error {
+
+	primaryDeviceClass := p.Replicated.HybridStorage.PrimaryDeviceClass
+	secondaryDeviceClass := p.Replicated.HybridStorage.SecondaryDeviceClass
+
+	err := validateDeviceClassOSDs(context, clusterInfo, primaryDeviceClass)
+	if err != nil {
+		return errors.Wrapf(err, "failed to validate primary device class %q", primaryDeviceClass)
+	}
+
+	err = validateDeviceClassOSDs(context, clusterInfo, secondaryDeviceClass)
+	if err != nil {
+		return errors.Wrapf(err, "failed to validate secondary device class %q", secondaryDeviceClass)
+	}
+
+	return nil
+}
+
+// validateDeviceClassOSDs validates that the device class should have atleast one OSD
+func validateDeviceClassOSDs(context *clusterd.Context, clusterInfo *cephclient.ClusterInfo,
+	deviceClassName string) error {
+	deviceClassOSDs, err := cephclient.GetDeviceClassOSDs(context, clusterInfo, deviceClassName)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get osds for the device class %q", deviceClassName)
+	}
+	if len(deviceClassOSDs) == 0 {
+		return errors.New(fmt.Sprintf("no osds available for the device class %q", deviceClassName))
```

Comment:
```suggestion
		return errors.New(fmt.Sprintf("no osds available for the device class %q", deviceClassName))
```

use `errors.Errorf()`